example/image-classification/symbols/resnet-v1-fp16.py

```diff
@@ -0,0 +1,196 @@
+        Channel size of each stage
+    num_classes : int
+        Ouput size of symbol
+    dataset : str
+        Dataset type, only cifar10 and imagenet supports
+    workspace : int
+        Workspace used in convolution operator
+    """
+    num_unit = len(units)
+    assert(num_unit == num_stages)
+    data = mx.sym.Variable(name='data')
+    data = mx.symbol.Cast(data=data, dtype=np.float16)
+    (nchannel, height, width) = image_shape
+    weight = mx.symbol.Variable(name='conv0_weight', dtype=np.float32)
+    weight = mx.symbol.Cast(data=weight, dtype=np.float16)
+    if height <= 32:            # such as cifar10
+        body = mx.sym.Convolution(data=data, weight=weight, num_filter=filter_list[0], kernel=(3, 3), stride=(1,1), pad=(1, 1),
+                                  no_bias=True, name="conv0", workspace=workspace)
+        # Is this BatchNorm supposed to be here?
+        body = mx.sym.BatchNorm(data=body, fix_gamma=False, eps=2e-5, momentum=bn_mom, name='bn0')
+    else:                       # often expected to be 224 such as imagenet
+        body = mx.sym.Convolution(data=data, weight=weight, num_filter=filter_list[0], kernel=(7, 7), stride=(2,2), pad=(3, 3),
+                                  no_bias=True, name="conv0", workspace=workspace)
+        body = mx.sym.BatchNorm(data=body, fix_gamma=False, eps=2e-5, momentum=bn_mom, name='bn0')
+        body = mx.sym.Activation(data=body, act_type='relu', name='relu0')
+        body = mx.symbol.Pooling(data=body, kernel=(3, 3), stride=(2,2), pad=(1,1), pool_type='max')
+
+    for i in range(num_stages):
+        body = residual_unit(body, filter_list[i+1], (1 if i==0 else 2, 1 if i==0 else 2), False,
+                             name='stage%d_unit%d' % (i + 1, 1), bottle_neck=bottle_neck, workspace=workspace,
+                             memonger=memonger)
+        for j in range(units[i]-1):
+            body = residual_unit(body, filter_list[i+1], (1,1), True, name='stage%d_unit%d' % (i + 1, j + 2),
+                                 bottle_neck=bottle_neck, workspace=workspace, memonger=memonger)
+    # bn1 = mx.sym.BatchNorm(data=body, fix_gamma=False, eps=2e-5, momentum=bn_mom, name='bn1')
+    # relu1 = mx.sym.Activation(data=bn1, act_type='relu', name='relu1')
+    # Although kernel is not used here when global_pool=True, we should put one
+    pool1 = mx.symbol.Pooling(data=body, global_pool=True, kernel=(7, 7), pool_type='avg', name='pool1')
+    flat = mx.symbol.Flatten(data=pool1)
+    weight = mx.symbol.Variable(name='fc1_weight', dtype=np.float32)
+    bias = mx.symbol.Variable(name='fc1_bias', dtype=np.float32)
+    weight = mx.symbol.Cast(data=weight, dtype=np.float16)
+    bias = mx.symbol.Cast(data=bias, dtype=np.float16)
+    fc1 = mx.symbol.FullyConnected(data=flat, weight=weight, bias=bias, num_hidden=num_classes, name='fc1')
+    label = mx.symbol.Variable(name='softmax_label')
+    label = mx.symbol.Cast(data=label, dtype=np.float16)
```

Comment:
Hmm.. I thought so too, but for some reason I thought I needed that. Maybe I don't?